plutus-tx/src/Language/PlutusTx/Evaluation.hs

```diff
@@ -0,0 +1,34 @@
+module Language.PlutusTx.Evaluation (evaluateCek, evaluateCekLog) where
+
+import           Language.PlutusCore.Interpreter.CekMachine hiding (evaluateCek)
+import           Language.PlutusCore
+import           Language.PlutusCore.Constant.Dynamic
+import           Language.PlutusCore.Constant
+
+import           Control.Exception
+
+import           System.IO.Unsafe
+
+stringBuiltins :: DynamicBuiltinNameMeanings
+stringBuiltins =
+    insertDynamicBuiltinNameDefinition dynamicCharToStringDefinition $ insertDynamicBuiltinNameDefinition dynamicAppendDefinition mempty
+
+-- TODO: pretty sure we shouldn't need the unsafePerformIOs here, we should expose a pure interface even if it has IO hacks under the hood
+
+-- | Evaluate a program in the CEK machine with the usual string dynamic builtins.
+evaluateCek
+    :: Program TyName Name ()
+    -> EvaluationResult
+evaluateCek p = unsafePerformIO $ evaluate $ runCek stringBuiltins p
```

Comment:
Is `unsafePerformIO . evaluate` not the same thing as `id`?

Reply:
Honestly I'm not sure. Is it? `evaluate` does some forcing.

Reply:
`evaluate` force evaluation to WHNF, but that'll happen anyway when this expression is evaluated; so, yes, it is `== id`.